nixos/modules/services/web-apps/filesender.nix

```diff
@@ -0,0 +1,403 @@
+                    description = "URL where SimpleSAMLphp can be reached.";
+                  };
+                };
+              };
+              default = { };
+              description = ''
+                Configuration options used by SimpleSAMLphp.
+                See [](https://simplesamlphp.org/docs/stable/simplesamlphp-install)
+                for available options.
+              '';
+            };
+
+            authSources = mkOption {
+              type = with types; phpSettingsType;
+              default = { };
+              example = { };
+              description = ''
+                Auth sources options used by SimpleSAMLphp.
+              '';
+            };
+          };
+        };
+    };
+
+    configureNginx = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Configure nginx as a reverse proxy for Filesender.";
+    };
+    localDomain = mkOption {
+      type = types.str;
+      example = "filesender.example.org";
+      description = "The domain serving your Filesender instance.";
+    };
+    poolSettings = mkOption {
+      type = with types; attrsOf (oneOf [ str int bool ]);
+      default = {
+        "pm" = "dynamic";
+        "pm.max_children" = "32";
+        "pm.start_servers" = "2";
+        "pm.min_spare_servers" = "2";
+        "pm.max_spare_servers" = "4";
+        "pm.max_requests" = "500";
+      };
+      description = ''
+        Options for Filesender's PHP pool. See the documentation on `php-fpm.conf` for details on configuration directives.
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    services.phpfpm = {
+      pools.filesender = {
+        user = cfg.user;
+        group = config.services.nginx.group;
+        inherit phpPackage;
+        phpEnv = {
+          FILESENDER_CONFIG_DIR = "${filesenderConfigDirectory}";
+          SIMPLESAMLPHP_CONFIG_DIR = "${simpleSamlPhpConfigDirectory}";
+        };
+        settings =
+          {
+            "listen.owner" = config.services.nginx.user;
+            "listen.group" = config.services.nginx.group;
+          }
+          // cfg.poolSettings;
+      };
+    };
+
+    services.nginx = lib.mkIf cfg.configureNginx {
+      enable = true;
+      virtualHosts.${cfg.localDomain} = {
+        root = "${cfg.package}/www";
+        extraConfig = ''
+          index index.php;
+        '';
+        locations = {
+          "/" = {
+            extraConfig = ''
+              try_files $uri $uri/ /index.php?args;
+            '';
+          };
+          "~ [^/]\\.php(/|$)" = {
+            extraConfig = ''
+              fastcgi_split_path_info  ^(.+\.php)(/.+)$;
+              fastcgi_pass  unix:${fpm.socket};
+              include ${pkgs.nginx}/conf/fastcgi.conf;
+              fastcgi_intercept_errors on;
+              fastcgi_param PATH_INFO       $fastcgi_path_info;
+              fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
+            '';
+          };
```

Comment:
I'm no expert, but this feels like it could conflict with other PHP software, no?

Reply:
It shouldn't because it's configured only for `nginx.virtrualHosts.${localDomain}`.